src/main/scala/com/jcdecaux/setl/transformation/FactoryDeliveryMetadata.scala

```diff
@@ -38,7 +38,7 @@ private[setl] case class FactoryDeliveryMetadata(factoryUUID: UUID,
    *
    * @return
    */
-  def getFactoryInputs: List[FactoryInput] = argTypes.map(tp => FactoryInput(tp, producer, id))
+  def getFactoryInputs(consumer: Class[_ <: Factory[_]]): List[FactoryInput] = argTypes.map(tp => FactoryInput(tp, producer, id, autoLoad, optional, consumer))
```

Comment:
Does the argument `consumer` always equal to the class of this factory? In this case maybe we add a field called `factoryClass` which is the class object of the factory that the delivery belongs to, for example: 
```scala
/**
 * DeliverySetterMetadata contains information of the @Delivery annotated method, including the name,
 * argument types, the producer and optional
 *
 * @param factoryUUID UUID of factory
 * @param factoryClass class of factory that has this delivery
 * @param symbol      symbol of the method
 * @param deliverySetter      either a field or a method
 * @param argTypes    type of each argument
 * @param producer    the producer class for this delivery
 * @param optional    true if this delivery is optional
 * @param autoLoad    true if this delivery should be automatically loaded
 * @param condition    string, filter condition that wil be applied during the auto-load
 * @param id    string, id of this delivery
 */
private[setl] case class FactoryDeliveryMetadata(factoryUUID: UUID,
                                                 factoryClass: Class[_ <: Factory[_]],
                                                 symbol: runtime.universe.Symbol,
                                                 deliverySetter: Either[Field, Method],
                                                 argTypes: List[runtime.universe.Type],
                                                 producer: Class[_ <: Factory[_]],
                                                 optional: Boolean,
                                                 autoLoad: Boolean = false,
                                                 condition: String = "",
                                                 id: String = Deliverable.DEFAULT_ID) {
...
}
```

And then we can keep the argument of this method empty. It's maybe more understandable